src/test/kotlin/com/igorwojda/string/ispalindrome/permutation/challenge.kt

```diff
@@ -4,7 +4,16 @@ import org.amshove.kluent.shouldBeEqualTo
 import org.junit.jupiter.api.Test
 
 private fun isPermutationPalindrome(str: String): Boolean {
-    TODO("not implemented")
+    fun isPalindrome(str: String) = str == str.reversed()
+    var permutation = str
+    for (i in str.indices) {
+        if (isPalindrome(permutation)) {
+            return true
+        } else {
+            permutation = permutation.substring(1, permutation.length) + permutation[0]
```

Comment:
Please don't confuse logical shifting with permutations. Add this test and see for yourself:
```
@Test
    fun `"okvok" is a palindrome`() {
        isPermutationPalindrome("okvok") shouldBeEqualTo true
    }
```